gnd/src/main/java/com/google/android/gnd/ui/offlinebasemap/viewer/OfflineBaseMapViewerViewModel.java

```diff
@@ -42,46 +48,64 @@
+  private final LiveData<OfflineBaseMap> offlineArea;
+  @Inject Navigator navigator;
 
   @Inject
   public OfflineBaseMapViewerViewModel(
-      OfflineBaseMapRepository offlineBaseMapRepository, @ApplicationContext Context context) {
+      OfflineBaseMapRepository offlineBaseMapRepository,
+      @ApplicationContext Context context,
+      Navigator navigator) {
     this.offlineBaseMapRepository = offlineBaseMapRepository;
     this.context = new WeakReference<>(context);
-    this.areaName =
-        LiveDataReactiveStreams.fromPublisher(
-            this.argsProcessor.switchMap(
+    this.navigator = navigator;
+    @Hot
+    // We only need to convert this single to a flowable in order to use it with LiveData.
+    // It still only contains a single offline area returned by getOfflineArea.
+    Flowable<OfflineBaseMap> offlineAreaItemAsFlowable =
+        this.fragmentArgs
+            .flatMap(
                 args ->
                     this.offlineBaseMapRepository
                         .getOfflineArea(args.getOfflineAreaId())
-                        .toFlowable()
-                        .map(OfflineBaseMap::getName)));
-    this.areaStorageSize =
+                        .doOnError(
+                            throwable ->
+                                Timber.e(
+                                    throwable,
+                                    "Couldn't render area: %s",
+                                    args.getOfflineAreaId())))
+            .toFlowable();
+    this.areaName =
         LiveDataReactiveStreams.fromPublisher(
-            this.argsProcessor.switchMap(
-                args ->
-                    this.offlineBaseMapRepository
-                        .getOfflineArea(args.getOfflineAreaId())
-                        .toFlowable()
-                        .flatMap(
-                            offlineBaseMapRepository
-                                ::getIntersectingDownloadedTileSourcesOnceAndStream)
-                        .map(this::tileSourcesToTotalStorageSize)));
-    this.offlineArea =
+            offlineAreaItemAsFlowable.map(OfflineBaseMap::getName));
+    this.areaStorageSize =
         LiveDataReactiveStreams.fromPublisher(
-            this.argsProcessor.switchMap(
-                args ->
-                    this.offlineBaseMapRepository
-                        .getOfflineArea(args.getOfflineAreaId())
-                        .toFlowable()));
+            offlineAreaItemAsFlowable
+                .flatMap(
+                    offlineBaseMapRepository::getIntersectingDownloadedTileSourcesOnceAndStream)
+                .map(this::tileSourcesToTotalStorageSize));
+    this.offlineArea = LiveDataReactiveStreams.fromPublisher(offlineAreaItemAsFlowable);
+    disposeOnClear(
+        removeAreaClicks
+            .first(Nil.NIL)
+            .flatMapCompletable(
+                __ -> {
+                  if (this.offlineArea.getValue() == null) {
```

Comment:
I believe using `area = Objects.requireNonNull(this.offlineArea.getValue(), "Could not remove.....");` would have the same effect here.

Reply:
Thanks! Unfortunately our min API is too low to use the form of that function that takes the string argument, but letting it throw the default null pointer exception and catching the error works just as well.